components/tools/OmeroPy/src/omero/plugins/ldap.py

```diff
@@ -163,7 +166,15 @@ def getdn(self, args):
         if dn is not None and dn.strip():
             self.ctx.out(dn)
 
-    @admin_only
+    @admin_only(AdminPrivilegeModifyGroup)
+    def update_group(self, iupdate, group):
+        iupdate.saveObject(group)
+
+    @admin_only(AdminPrivilegeModifyUser)
+    def update_user(self, iupdate, user):
+        iupdate.saveObject(user)
+
+    @admin_only()
```

Comment:
:+1: Maybe just depends if we want to handle these conditional checks via decorated internal methods or by the API also offering `check_admin`.